### ISSUE TITLE
RPM package for OpenTelemetry library

### ISSUE THREAD

**Issue by sirzooro** [open-telemetry/opentelemetry-cpp#1752]
I need to build rpm package for OpenTelemetry library. I found that cmake is able to generate it too. To make it work, I added following lines at the end of CMakeLists.txt:

```
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/usr/local")
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/usr/local/include")
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/usr/local/lib64")
list(APPEND CPACK_RPM_EXCLUDE_FROM_AUTO_FILELIST_ADDITION "/usr/lib64/cmake")
set(CPACK_PACKAGE_VERSION ${OPENTELEMETRY_VERSION})
set(CPACK_GENERATOR "RPM")
set(CPACK_PACKAGE_NAME "opentelemetry-cpp")
set(CPACK_PACKAGE_RELEASE 1)
set(CPACK_PACKAGE_CONTACT "The OpenTelemetry Authors")
set(CPACK_PACKAGE_VENDOR "The OpenTelemetry Authors")
set(CPACK_PACKAGING_INSTALL_PREFIX ${CMAKE_INSTALL_PREFIX})
set(CPACK_PACKAGE_FILE_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${CPACK_PACKAGE_RELEASE}.${CMAKE_SYSTEM_PROCESSOR}")
include(CPack)
```

With this change I am able to create rpm package using `make package` command. Please add this to your CMakeLists.txt (or maybe CMakeSettings.json?).

I also noticed that generated rpm package contains `libhttp_client_nosend.so` library. This library does not follow library name convention, it should be named `libopentelemetry_http_client_nosend.so`. Additionally it is linked with gtest and gmock library, so I wonder if it should go into final package at all. Not a big deal for me, but you may want to clean this up too.

**Comment by lalitb** [comment]
> I need to build rpm package for OpenTelemetry library. 

@sirzooro - If you want to check the PR #1662, to enable generating both deb and rpm packages. I have tested it for deb, if you want to test it for rpm too.

> Additionally it is linked with gtest and gmock library, so I wonder if it should go into final package at all. Not a big deal for me, but you may want to clean this up too.

Yes, have a issue #1226 to track this. Just not getting prioritized with metrics and logging work going on.

> I also noticed that generated rpm package contains libhttp_client_nosend.so library. This library does not follow library name convention, it should be named libopentelemetry_http_client_nosend.so

Agree.

**Comment by sirzooro** [comment]
Thanks, I will check that PR and add some comments.

**Comment by lalitb** [comment]
closed with #1662
